src/core_scan/Core_scan.mli

```diff
@@ -121,6 +121,24 @@ val output_core_results :
     (depending on the value in config.output_format)
 *)
 
+(*****************************************************************************)
+(* Pre and Post Processors Hook For Semgrep Pro / Extensions        *)
+(*****************************************************************************)
+
+module type Pre_and_post_processor = sig
```

Comment:
All of that could be in a separate file.
I don't like nested modules, usually it's as sign of bad separation of concerns.